ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/WithdrawalsValidatorProvider.java

```diff
@@ -15,17 +15,22 @@
 
 package org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.engine;
 
+import org.hyperledger.besu.ethereum.core.BlockHeader;
 import org.hyperledger.besu.ethereum.mainnet.ProtocolSpec;
 import org.hyperledger.besu.ethereum.mainnet.TimestampSchedule;
 import org.hyperledger.besu.ethereum.mainnet.WithdrawalsValidator;
 
+import java.util.Optional;
+
 public class WithdrawalsValidatorProvider {
 
   static WithdrawalsValidator getWithdrawalsValidator(
-      final TimestampSchedule timestampSchedule, final long newPayloadTimestamp) {
+      final TimestampSchedule timestampSchedule,
+      final BlockHeader parentBlockHeader,
+      final long newPayloadTimestamp) {
 
-    return timestampSchedule
-        .getByTimestamp(newPayloadTimestamp)
+    return Optional.ofNullable(
+            timestampSchedule.getForNextBlockHeader(parentBlockHeader, newPayloadTimestamp))
```

Comment:
Using getForNextBlockHeader makes sense for forkchoiceUpdated.

For newPayload, it's a bit dubious due to the parentHeader change in behaviour mentioned above. 
An alternative for newPayload would be to use getByBlockHeader and construct a dummy BlockHeader instead, which would require a second version of getWithdrawalsValidator.

We can of course use the BlockHeader that is constructed later on in newPayload, however it's a bit of a chicken and egg because it takes the withdrawals as a parameter and also required changing the order that we respond with INVALID in newPayload which I wanted to avoid if possible. Think a dummy header is preferable over this, since we can't guarantee validity of the whole header at this point.

Reply:
Agree that for FCU it makes sense to use getForNextBlockHeader.

For newPayload would be better to use the actual header we construct, but we could get away with a dummy header. If we are using a dummy header though using getForNextBlockHeader with the new timestamp and chain head would also work since we only use the timestamp from that block in the check.

Reply:
As discussed, plan to use getForNextBlockHeader for FCU and getByBlockHeader(dummyHeader) for newPayload.